src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_evp.c

```diff
@@ -57,6 +57,43 @@ int32_t CryptoNative_EvpDigestFinalEx(EVP_MD_CTX* ctx, uint8_t* md, uint32_t* s)
     return ret;
 }
 
+EVP_MD_CTX* CryptoNative_EvpDup(const EVP_MD_CTX* ctx)
+{
+    if (ctx == NULL)
+    {
+        return NULL;
+    }
+
+    EVP_MD_CTX* dup = EVP_MD_CTX_new();
+
+    if (dup == NULL)
+    {
+        return NULL;
+    }
+
+    if (!EVP_MD_CTX_copy(dup, ctx))
```

Comment:
`EVP_MD_CTX_copy` is obsolete. Consider `EVP_MD_CTX_copy_ex`? IIRC it goes back far enough support-wise.

Reply:
What notion of obsolete do you see? The docs say it just works on non-initialized contexts.

https://www.openssl.org/docs/man1.1.1/man3/EVP_MD_CTX_copy.html

> EVP_MD_CTX_copy()
>    Similar to EVP_MD_CTX_copy_ex() except the destination out does not have to be initialized.